src/render/painter.js

```diff
@@ -177,6 +179,25 @@ class Painter {
         terrain.update(style, this.transform, cameraChanging);
     }
 
+    _updateFog() {
+        if (this.transform.pitch <= FOG_PITCH_END || !(this.style && this.style.fog)) {
+            this.transform.fogCullDistSq = null;
+            return;
+        }
+
+        const fog = this.style.fog;
+        const fogStart = fog.properties.get('range')[0];
+        const fogEnd = fog.properties.get('range')[1];
+        const fogDensity = fog.properties.get('density');
+        // We start culling where the fog opacity function hits 98%, leaving
+        // a non-noticeable opacity change threshold. We use an arbitrary function
+        // which bounds the true answer. See: https://www.desmos.com/calculator/lw03ldsuhy
+        const fogBoundFraction = 1 - 0.22 * Math.exp(4 * (fogDensity - 1));
+        const fogCullDist = fogStart + (fogEnd - fogStart) * fogBoundFraction;
```

Comment:
This corresponds to a shader function, right? might be worth explicitly naming that here

Reply:
No, this one doesn't have any shader function associated with it. It's finding a safe bound between start and end where we can safely start cutting off tiles.

We can fix the naming of that one to be more explicit.